src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -826,6 +835,44 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType, bool inExpressionTr
                 refKind,
                 returnType);
 
+        private LambdaSymbol? _lazyLambdaSymbol = null;
+
+        internal LambdaSymbol? LazyInProgressLambdaSymbol
+        {
+            get
+            {
+                if (_lazyLambdaSymbol is null)
+                {
+                    Interlocked.CompareExchange(ref _lazyLambdaSymbol, CreateTemporaryLambdaSymbol(), null);
+                }
+
+                return _lazyLambdaSymbol;
+            }
+        }
+
+        internal LambdaSymbol? CreateTemporaryLambdaSymbol()
```

Comment:
>CreateTemporaryLambdaSymbol [](http://example.com/codeflow?start=31&length=27)

Consider making this a local function in the property above, and perhaps just named `createLambda()`.